### DISCUSSION TITLE
Rename BRiM to SymBRiM

### DISCUSSION THREAD

**Discussion by tjstienstra** [mechmotum/symbrim#140] [General]
Already for some time I have been thinking about renaming BRiM. Therefore, it deemed me best to open a discussion on it.

The current name of the package is _BRiM_: **B**icycle **Ri**der **M**odels.
The new name I would like to propose is _SymBRiM_: **Sym**bolic **B**icycle **Ri**der **M**odels.

There are several reasons why I would like to rename `BRiM` to `SymBRiM`:
- The word BRiM is also used in different contexts as a word and an abbreviation, both of which are impractical occasionally.
- SymBRiM is a (more) unique name.
- The name also seems to make releasing it more difficult, refer to #37.
- SymBRiM is more complete, as it also covers the fact that we are focusing on deriving the symbolic equations of motion.
- SymBRiM has two syllables, which is also giving a practical alias `sb`. As far as I know, there is no other package using this alias (seaborn uses `sns`).

There are also a few disadvantages of renaming:
- The package has been published under the name BRiM in the scientific community, i.e. the [BRiM paper](https://doi.org/10.59490/6504c5a765e8118fc7b106c3)
- Though redirects can be set up (mostly automatically if done right), they can be broken.
- Takes some effort, but I'm willing to do it.

@moorepants what is your opinion on this?

**Comment by moorepants** [comment]
I'm fine with this. It would really be nice to get in on PyPi and Conda Forge for easy installation.

There is the detail that the PyPi "distribution" can be different from the one or more package names. So you can retain `import brim` but you'd type `pip install symbrim`. Not sure if that is desired.

Another option: BRiMs for Models being plural?

**Reply by tjstienstra** [reply]
> I'm fine with this. It would really be nice to get in on PyPi and Conda Forge for easy installation.

Thank you for the quick response. I'll see if I can make the adjustment somewhere this week.

> There is the detail that the PyPi "distribution" can be different from the one or more package names. So you can retain import brim but you'd type pip install symbrim. Not sure if that is desired.

I prefer having the same name on pypi as the package. I always find it confusing when a package has a different name on PyPi, makes me having to search.

> Another option: BRiMs for Models being plural?

If we're going with a rename, I'd rather choose the most optimal one and I think that `SymBRiM` is more unique, complete, and has a sensible alias `sb`.

**Reply by moorepants** [reply]
SymBRiM has a nice verbal ring to it.
